### DISCUSSION TITLE
error in converting conllu to spacy files

### DISCUSSION THREAD

**Discussion by toufiglu** [explosion/spaCy#11653] [Help: Coding & Implementations] [answered]
Hi. I was trying to convert a dev treebank file to spacy format to prepare for training. I got the following error:
```
ValueError: [E190] Token head out of range in `Doc.from_array()` for token index '19' with value '2' 
(equivalent to relative head index: '2'). 
The head indices should be relative to the current token index rather than absolute indices in the array.
```
I checked whether there was a token which was the 19th of a sentence, and saw whether its head was 2. I eliminated the two instances when I found them, and run an easy python script to check that. The error remains. I wonder what I can do to solve this.

Thank you,

**Comment by adrianeboyd** [comment, marked as answer]
Are you using the built-in CoNLL-U converter?

If so, have you validated your CoNLL-U input with the UD tools? You can validate the format with the UD validation tools (https://github.com/universaldependencies/tools) with `validate.py --level 2` for the CoNLL-U + basic UD validation only. The output might include errors related to non-UD deprels if you have any of those, but you can ignore those errors and focus on errors related to ID/HEAD values.

**Reply by toufiglu** [reply]
Hi Adriane,

Thanks. I followed the step and the file is converted now. I am grateful for your answer!
